File: benchmark/interactive_function/Metrix_multiplication/Metrix.js
'use strict';
const n = 1000;
const a = new Float64Array(n * n);
const b = new Float64Array(n * n);
const res = new Float64Array(n * n);

for (let i = 0; i < n * n; i++) {
    a[i] = i % n;
    b[i] = Math.floor(i / n);
}

console.log(`Starting Matrix Multiplication (Node.js - Flat): ${n}x${n}`);
const start = performance.now();

for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
        let sum = 0;
        for (let k = 0; k < n; k++) {
            sum += a[i * n + k] * b[k * n + j];
        }
        res[i * n + j] = sum;
    }
}

const end = performance.now();
console.log(`Sample Result [0]: ${res[0]}`);
console.log(`Time: ${((end - start) / 1000).toFixed(4)} sec`);

File: benchmark/interactive_function/Metrix_multiplication/Metrix.go
package main

import (
	"fmt"
)

func main() {
	const n = 1000
	a := make([]float64, n*n)
	b := make([]float64, n*n)
	res := make([]float64, n*n)

	for i := 0; i < n*n; i++ {
		a[i] = float64(i % n)
		b[i] = float64(i / n)
	}

	fmt.Printf("Starting Matrix Multiplication (Go - Flat): %dx%d\n", n, n)
	// start := time.Now()

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			var sum float64
			for k := 0; k < n; k++ {
				sum += a[i*n+k] * b[k*n+j]
			}
			res[i*n+j] = sum
		}
	}

	// duration := time.Since(start)
	fmt.Printf("Sample Result [0]: %f\n", res[0])
	// fmt.Printf("Time: %v\n", duration.Seconds())
}
